File: 0202_Happy Number/HappyNumber.cs
public class Solution {
    public bool IsHappy(int n) {
        if(n <= 0) return false;
        if(n == 1) return true;
        var set = new HashSet<int>();
        
        while(!set.Contains(n))
        {
            set.Add(n);
            var sum = 0;
            while(n > 0)
            {
                sum += (n % 10) * (n % 10);
                n /= 10;
            }
            
            if(sum == 1) return true;
            n = sum;
        }
        
        return false;
    }
}

File: 0202_Happy Number/HappyNumber.java
class Solution {
    public boolean isHappy(int n) {
        Set<Integer> set = new HashSet<>();
        while(n != 1){
            n = convert(n);
            if(!set.add(n)) return false;
        }
        
        return true;
    }
    
    private int convert(int n){
        int v = 0;
        while(n > 0){
            v += (n % 10) * (n % 10);
            n /= 10;
        }
        return v;
    }
}